lib/travis/build/script/dart.rb

```diff
@@ -88,60 +150,124 @@ def announce
 
         def install
           sh.if '-f pubspec.yaml' do
-            sh.cmd "pub get"
+            sh.fold 'pub_get' do
+              sh.cmd "pub get"
+            end
           end
         end
 
         def script
           # tests with test package
-          sh.if '[[ -d packages/test ]] || grep -q ^test: .packages 2> /dev/null', raw: true do
-            if with_content_shell
+          sh.if package_installed?('test'), raw: true do
+            if config[:with_content_shell]
               sh.export 'DISPLAY', ':99.0'
               sh.cmd 'sh -e /etc/init.d/xvfb start'
               # give xvfb some time to start
               sh.cmd 't=0; until (xdpyinfo -display :99 &> /dev/null || test $t -gt 10); do sleep 1; let t=$t+1; done'
               sh.cmd 'pub run test -p vm -p content-shell -p firefox'
             else
-              sh.cmd 'pub run test'
+              pub_run_test
             end
           end
           # tests with test_runner for old tests written with unittest package
-          sh.elif '[[ -d packages/unittest ]] || grep -q ^unittest: .packages 2> /dev/null', raw: true do
+          sh.elif package_installed?('unittest'), raw: true do
+            sh.fold "deprecated.unittest" do
+              sh.deprecate <<MESSAGE
+DEPRECATED: The unittest package is deprecated. Please upgrade to the test
+package. See https://github.com/dart-lang/test#readme.
+MESSAGE
+            end
+
             sh.fold 'test_runner_install' do
               sh.echo 'Installing Test Runner', ansi: :yellow
               sh.cmd 'pub global activate test_runner'
             end
 
-            if with_content_shell
+            if config[:with_content_shell]
               sh.cmd 'xvfb-run -s "-screen 0 1024x768x24" pub global run test_runner --disable-ansi'
             else
               sh.cmd 'pub global run test_runner --disable-ansi --skip-browser-tests'
             end
           end
+
+          dartanalyzer
+          dartfmt
         end
 
         private
 
+          def pub_run_test
+            args = task[:test]
+            unless args
+              sh.raw ':'
+              return
```

Comment:
Doesn't it [insert the s-expression into a list](https://github.com/travis-ci/travis-build/blob/master/lib/travis/shell/builder.rb#L38) rather than returning it?